### PR TITLE
Enhacements Breadcrumbs, update docs.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Clean code.
- [x] Adjust Breadcrumbs template with icons. https://bulma.io/documentation/components/breadcrumb/#icons
- [x] Add more tests.
- [x] Update docs.

### CODE REVIEW

**Comment by samdark** [comment]
Should we adjust demo somehow?

**Comment by terabytesoftw** [comment]
> ¿Deberíamos ajustar la demo de alguna manera?

> Should we adjust demo somehow?

The demo has Bootstrap5 and the app doesn't use it.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Breadcrumbs.php [hunk 1]
```diff
@@ -29,23 +27,89 @@
...
+        $new->autoIdPrefix = $value;
+        return $new;
+    }
+
+    /**
+     * Set encode to true to encode the output.
+     *
+     * @param bool $value whether to encode the output.
+     *
+     * @return $this
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:107]: done.

File: src/Breadcrumbs.php [hunk 2]
```diff
@@ -29,23 +27,89 @@
...
+    private array $itemsAttributes = [];
+    private string $itemTemplate = "<li>{link}</li>\n";
 
     /**
-     * Disables encoding for labels and returns a new instance.
+     * Defines a string value that labels the current element.
+     *
+     * @param string $value The value of the aria-label attribute.
+     *
+     * @return static
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:44]: done.

File: src/Breadcrumbs.php [hunk 3]
```diff
@@ -29,23 +27,89 @@
...
+        $new->attributes = $value;
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified prefix to the automatically generated widget IDs.
+     *
+     * @param string $value The prefix to the automatically generated widget IDs.
+     *
+     * @return static
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:91]: done.

File: src/Breadcrumbs.php [hunk 4]
```diff
@@ -74,32 +138,16 @@ public function homeItem(?array $value): self
...
     /**
-     * Returns a new instance with the specified item template.
+     * Returns a new instance with the specified ID of the widget.
      *
-     * @param string $value The template used to render each inactive item in the breadcrumbs.
-     * The token `{link}` will be replaced with the actual HTML link for each inactive item.
+     * @param string $value The ID of the widget.
      *
-     * @return self
+     * @return static
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:145]: done.

File: src/Breadcrumbs.php [hunk 5]
```diff
@@ -272,4 +306,22 @@ private function renderItems(): array
 
         return $items;
     }
+
+    /**
+     * Set id of the widget.
+     *
+     * @return static
+     */
+    private function setId(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:315]: done.

File: src/Breadcrumbs.php [hunk 6]
```diff
@@ -127,75 +175,63 @@ public function items(array $value): self
...
+        $new = clone $this;
+
+        if (empty($new->items)) {
             return '';
         }
 
-        $this->buildOptions();
+        $new = $new->setId();
+        $new = $new->ariaLabel('breadcrumbs');
+        Html::addCssClass($new->attributes, 'breadcrumb');
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:218]: done.

File: src/Breadcrumbs.php [hunk 7]
```diff
@@ -272,4 +306,22 @@ private function renderItems(): array
 
         return $items;
     }
+
+    /**
+     * Set id of the widget.
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php]: done.

File: src/Breadcrumbs.php [hunk 8]
```diff
@@ -206,54 +242,51 @@ private function renderIcon(string $icon, array $iconOptions): string
...
+        unset($item['label']);
 
-        if (isset($item['url'])) {
-            $options = $item;
-            unset($options['template'], $options['label'], $options['url'], $options['icon']);
-            $link = Html::a($label, $item['url'], $options)->encode(false)->render();
-        } else {
-            $link = $label;
+        if ($icon !== '') {
+            $label = $icon . '<span>' . $label . '</span>';
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php:286]: https://bulma.io/documentation/components/breadcrumb/#icons it explicitly goes inside a <span> tag.

[the remark this answers is not in the document]
Comment (samdark) [reply src/Breadcrumbs.php:286]: Yes. But later you apply encoding and it will encode `<span>`.

[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php]: fixed, i use `Span::tag()`.

File: src/Breadcrumbs.php [hunk 9]
```diff
@@ -127,34 +179,33 @@ public function items(array $value): self
     }
 
     /**
-     * Returns a new instance with the specified options.
+     * Returns a new instance with the specified item template.
```
Comment (samdark) [inline src/Breadcrumbs.php:182]:
```suggestion
     * Returns a new instance with the specified item HTML attributes.
```

**Comment by samdark** [comment]
👍

**Pull request merged**